src/Bundle/CoverallsBundle/Collector/CiEnvVarsCollector.php

```diff
@@ -110,6 +111,47 @@ protected function fillTravisCi()
         return $this;
     }
 
+    /**
+     * Fill Github Actions environment variables.
+     *
+     * @return $this
+     */
+    protected function fillGithubActions()
+    {
+        if (!isset($this->env['GITHUB_ACTIONS'])) {
+            return $this;
+        }
+        $this->env['CI_NAME'] = 'github';
+
+        $githubEventName = $this->env['GITHUB_EVENT_NAME'];
+        $githubSha = $this->env['GITHUB_SHA'];
+        $githubRef = $this->env['GITHUB_REF'];
+
+        if (strpos($githubRef, 'refs/heads/') !== false) {
+            $githubRef = str_replace('refs/heads/', '', $githubRef);
+            $jobId = $githubSha;
+        } elseif ($githubEventName === 'pull_request') {
+            $refParts = explode('/', $githubRef);
+            $prNumber = $refParts[2];
+            $this->env['CI_PULL_REQUEST'] = $prNumber;
+            $this->readEnv['CI_PULL_REQUEST'] = $this->env['CI_PULL_REQUEST'];
+            $jobId = sprintf('%s-PR-%s', $githubSha, $prNumber);
+        } elseif (strpos($githubRef, 'refs/tags/') !== false) {
+            $githubRef = str_replace('refs/tags/', '', $githubRef);
+            $jobId = $githubRef;
+        }
+
+        $this->env['CI_JOB_ID'] = $jobId;
+        $this->env['CI_BRANCH'] = $githubRef;
```

Comment:
Referential implementation did not send this if I remember correctly. Currently I'm using it without setting branch. 

The problem is that `$githubRef` is not always branch so I'm not sure this is correct.

Reply:
I would itsn't principal moment